official/vision/beta/projects/yolo/losses/yolo_loss.py

```diff
@@ -0,0 +1,654 @@
+    Args: 
+      true_counts: `Tensor` of shape [batchsize, height, width, num_anchors] 
+        represeneting how many boxes are in a given pixel [j, i] in the 
+        output map.
+      inds: `Tensor` of shape [batchsize, None, 3] indicating the location 
+        [j, i] that a given box is associatied with in the FPN prediction 
+        map. 
+      y_true: `Tensor` of shape [batchsize, None, 8] indicating the actual box 
+        associated with each index in the inds tensor list.
+      boxes: `Tensor` of shape [batchsize, None, 4] indicating the original 
+        ground truth boxes for each image as they came from the decoder used
+        for bounding box search. 
+      classes: `Tensor` of shape [batchsize, None, 1] indicating the original 
+        ground truth classes for each image as they came from the decoder used
+        for bounding box search.
+      y_pred: `Tensor` of shape [batchsize, height, width, output_depth] 
+        holding the models output at a specific FPN level.
+
+    Return:
+      loss: `float` for the actual loss.
+      box_loss: `float` loss on the boxes used for metrics.
+      conf_loss: `float` loss on the confidence used for metrics.
+      class_loss: `float` loss on the classes used for metrics.
+      avg_iou: `float` metric for the average iou between predictions 
+        and ground truth.
+      avg_obj: `float` metric for the average confidence of the model 
+        for predictions.
+    """
+    (loss, box_loss, conf_loss, class_loss, mean_loss, iou, pred_conf, ind_mask,
+     grid_mask) = self.call(true_counts, inds, y_true, boxes, classes, y_pred)
+
+    # Temporary metrics
+    box_loss = tf.stop_gradient(0.05 * box_loss / self._iou_normalizer)
+
+    # Metric compute using done here to save time and resources.
+    sigmoid_conf = tf.stop_gradient(tf.sigmoid(pred_conf))
+    iou = tf.stop_gradient(iou)
+    avg_iou = loss_utils.avgiou(
+        loss_utils.apply_mask(tf.squeeze(ind_mask, axis=-1), iou))
+    avg_obj = loss_utils.avgiou(tf.squeeze(sigmoid_conf, axis=-1) * grid_mask)
+    return (loss, box_loss, conf_loss, class_loss, mean_loss,
+            tf.stop_gradient(avg_iou), tf.stop_gradient(avg_obj))
+
+  @abc.abstractmethod
+  def _build_per_path_attributes(self):
+    """Additional initialization required specifically for each unique YOLO 
+    loss version"""
+    ...
+
+  @abc.abstractmethod
+  def call():
+    """The actual logic to apply to the raw model for optimization."""
+    ...
+
+  def post_path_aggregation(self, loss, ground_truths, predictions):
+    """This method allows for post processing of a loss value after the loss
+    has been aggregateda across all the FPN levels."""
+    return loss
+
+  @abc.abstractmethod
+  def cross_replica_aggregation(self, loss, num_replicas_in_sync):
+    """This controls how the loss should be aggregated across replicas."""
+    ...
+
+
+@tf.custom_gradient
+def grad_sigmoid(values):
+  # This is an identity operation that will
```

Comment:
Explain first what this function does, then explain how it can be used later.

Reply:
added an explanation